libguard/guard_interface.cpp

```diff
@@ -40,11 +40,19 @@ void initialize()
     {
         if (!fs::exists(GUARD_PRSV_PATH))
         {
-            guard_log(GUARD_ERROR, "Fail to find guard file %s",
-                      GUARD_PRSV_PATH);
+            std::string exceptionLog = "Guard file " +
+                                       std::string(GUARD_PRSV_PATH) +
+                                       " does not exist";
+            throw InvalidGuardFile(exceptionLog);
         }
         guardFilePath = GUARD_PRSV_PATH;
     }
+    if (fs::file_size(guardFilePath) == 0)
+    {
+        std::string exceptionLog =
```

Comment:
JS: You could use stringstream or string constructor.

Reply:
agree you can change to (1 constructor and operator +)
    std::string str("Empty Guard file ");
    str+=GUARD_PRSV_PATH;